labman/db/tests/tester.py

```diff
@@ -224,6 +229,175 @@ def integration_tests():
             'Shotgun sample sheet does not match expected regex:\n%s' % obs)
 
 
+def stress_tests_amplicon_workflow(user, samples, num_plates=1):
+    print('Amplicon workflow', flush=True)
+    # Sample Plating
+    sp_processes = []
+    sample_iter = cycle(samples)
+    print('\tSample plating process', end='', flush=True)
+    for i in range(num_plates):
+        plate_samples = [next(sample_iter) for i in range(84)]
+        sp_processes.append(create_sample_plate_process(user, plate_samples))
+        print('.', end='', flush=True)
+    print('Ok', flush=True)
+
+    # gDNA extraction
+    print('\tgDNA extraction process', end='', flush=True)
+    ext_processes = []
+    for _, sp in sp_processes:
+        ext_processes.append(create_gdna_extraction_process(user, sp))
+        print('.', end='', flush=True)
+    print('Ok', flush=True)
+
+    # Amplicon library prep
+    print('\tAmplicon library prep process', end='', flush=True)
+    amplicon_processes = []
+    for _, gp in ext_processes:
+        amplicon_processes.append(create_amplicon_prep(user, gp))
+        print('.', end='', flush=True)
+    print('Ok', flush=True)
+
+    # Library plate quantification
+    print('\tAmplicon library quantification process', end='', flush=True)
+    amplicon_quant_processes = []
+    for _, ap in amplicon_processes:
+        amplicon_quant_processes.append(
+            create_quantification_process(user, ap))
+        print('.', end='', flush=True)
+    print('Ok', flush=True)
+
+    # Plate pooling process
+    print('\tPlate pooling process', end='', flush=True)
+    plate_pool_processes = []
+    for aqp, (_, ap) in zip(amplicon_quant_processes, amplicon_processes):
+        plate_pool_processes.append(
+            create_plate_pool_process(
+                user, aqp, ap,
+                {'function': 'amplicon',
+                 'parameters': {"dna_amount": 240, "min_val": 1, "max_val": 15,
+                                "blank_volume": 2, "robot": 6,
+                                "destination": 1}}))
+        print('.', end='', flush=True)
+    print('Ok', flush=True)
+
+    # Quantify pools
+    print('\tPlate pool quantification process', end='', flush=True)
+    pool_quant_processes = []
+    for pos in range(0, num_plates, 8):
```

Comment:
It looks like this is trying to pool sets of 8 amplicon pools per sequencing process. Should the logic be more like on line 333, where you're adding in just the set of 8 pools in this increment?

Reply:
The pool quantification process is done in batch, rather than having a 1-1 relationship (matches the nature of the process in the wet lab) so it is grouping it here and providing that information to the `create_pools_pool_process` function.

Reply:
but looking at the logic, isn't the case that `pools` here is going to be the entire set of N plates in `plate_pool_processes` for each `pool_quant_process`, when really it should just be the next set of 8?

Reply:
Nope, this for loop is grouping the pools in sets of 8 to create a single sequencing pool. Then each of this individual sequencing pools are individually "sequenced".

Reply:
I just talked with @ElDeveloper and I saw the bug on the code - I will push the fix soon.